src/index/IndexImpl.cpp

```diff
@@ -454,10 +455,17 @@ IndexBuilderDataAsExternalVector IndexImpl::passFileForVocabulary(
   // batch of triples and partial vocabulary.
   std::array<std::future<void>, 3> writePartialVocabularyFuture;
 
+  // Have a nice progress bar that shows the number of triples parsed. Depending
+  // on how the input is generated, it may take a long time before the first
+  // triple is parsed. We do not want to count that time, hence we stop the
+  // timer at this point and start it again when the first triple is parsed.
```

Comment:
What this comment doesn't explain is why the first triple should not be considered by the progress bar. I assume it's because the first triple also does some initialisation, which is why it is parsed slower than the others but that doesn't explain why it doesn't make sense for the progress bar to include it.